### PR TITLE
chore: refactor rejection of generic shared functions

### PR DESCRIPTION
**Pull request by crusso**
* minor refactoring of #3522

### CODE REVIEW

File: src/mo_frontend/typing.ml [hunk 1]
```diff
@@ -467,10 +476,8 @@ and check_typ' env typ : T.typ =
     let ts2 = List.map (check_typ env') typs2 in
     check_shared_return env typ2.at sort.it c ts2;
     if Type.is_shared_sort sort.it then
-    if not env.pre then begin
-      if List.length binds <> 1 then
-        error env typ.at "M0180"
-          "shared function has unexpected type parameters";
+      if not env.pre then begin
```
Comment (ggreif) [inline src/mo_frontend/typing.ml:479]: why not `&&`?

File: Changelog.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Motoko compiler changelog
 
+  * Fix bug in `Buffer` library affecting `filterEntries` (dfinity/motoko-base#422).
```
Comment (ggreif) [inline Changelog.md:3]: this is a `motoko-base` entry, right?

Comment (ggreif) [reply Changelog.md:3]: already mentioned below?

Comment (ggreif) [reply Changelog.md:3]: oh, another one?

Comment (crusso) [reply Changelog.md:3]: Yes, and good catch

**Comment by crusso** [comment]
@ggreif  PTAL

**Review by ggreif — APPROVED** [review]
LGTM, thanks!

**Pull request merged**